Session/Home/HomeViewModel.swift

```diff
@@ -578,16 +617,44 @@ public class HomeViewModel: NavigatableStateHolder {
         dependencies[defaults: .standard, key: .rateAppRetryAttemptCount] = 0
         
         if let scene = UIApplication.shared.connectedScenes.first(where: { $0.activationState == .foregroundActive }) as? UIWindowScene {
-            SKStoreReviewController.requestReview(in: scene)
+            NotificationCenter.default.addObserver(
+                self,
+                selector: #selector(self.systemHasPresentedNewWindow(notification:)),
+                name: UIWindow.didBecomeVisibleNotification, object: nil
+            )
+            
+            if !dependencies[feature: .simulateAppReviewLimit] {
+                SKStoreReviewController.requestReview(in: scene)
+            }
+
+            // Added 2 sec delay to give time for requet review to proc
+            DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(2)) { [self] in
+                NotificationCenter.default.removeObserver(self, name: UIWindow.didBecomeVisibleNotification, object: nil)
+                
+                guard didPresentAppReviewPrompt else {
+                    // Show rate limit prompt
+                    handlePromptChangeState(.rateLimit)
+                    return
+                }
+                
+                // Reset flag just in case it will be triggered again
+                didPresentAppReviewPrompt = false
+            }
         }
     }
     
+    @objc
+    func systemHasPresentedNewWindow(notification: Notification) {
```

Comment:
Can you please rename this to `windowBecameVisibleAfterTriggeringAppStoreReview` (the current name is very generic but the functionality is tied to the review process so the name should probably be specific)